src/backend/utils/adt/json.c

```diff
@@ -150,14 +152,28 @@ json_categorize_type(Oid typoid,
 	/* Look through any domain */
 	typoid = getBaseType(typoid);
 
+	if (sql_dialect==SQL_DIALECT_TSQL)
+	{
+		switch (typoid)
+		{
+			/* 
+			 * These OID are specifically for babelfish and therefore no macro is defined.
+			 * This switch case is to handle Bit dataype, to convert the typoid to BOOLOID.
+			 */
+			case 17056:
```

Comment:
We cannot use a hard-coded oid because it is not a fixed number.

Reply:
@2jungkook do you know of a way to get the Babelfish OID from within the engine code? I was wondering about this as well

Reply:
You may need to introduce a new hook, e.g., `lookup_tsql_datatype_oid_hook` (Ref: typecode.c).